packages/ra-ui-materialui/src/input/SelectInput.js

```diff
@@ -129,122 +131,119 @@ const styles = theme =>
  * <SelectInput source="gender" choices={choices} disableValue="not_available" />
  *
  */
-export class SelectInput extends Component {
+export const SelectInput = ({
+    allowEmpty,
+    choices,
+    classes,
+    className,
+    disableValue,
+    emptyValue,
+    helperText,
+    input,
+    isRequired,
+    label,
+    meta,
+    options,
+    optionText,
+    optionValue,
+    resource,
+    source,
+    translateChoice,
+    ...rest
+}) => {
     /*
      * Using state to bypass a redux-form comparison but which prevents re-rendering
      * @see https://github.com/erikras/redux-form/issues/2456
      */
-    state = {
-        value: this.props.input.value,
-    };
-
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.input.value !== this.props.input.value) {
-            this.setState({ value: nextProps.input.value });
-        }
-    }
-
-    handleChange = eventOrValue => {
-        const value = eventOrValue.target
-            ? eventOrValue.target.value
-            : eventOrValue;
-        this.props.input.onChange(value);
+    const [value, setValue] = useState(input.value);
+    const translate = useTranslate();
 
-        // HACK: For some reason, redux-form does not consider this input touched without calling onBlur manually
-        this.props.input.onBlur(value);
-        this.setState({ value });
-    };
+    useEffect(() => {
```

Comment:
Obviously a code smell. This is the implementation of the previous `componentWillReceiveProps`. Hopefully this will not be needed after migrating to final-forms